### ISSUE TITLE
Different Behaviour for ChildLifetimeScopeBeginning Event

### ISSUE THREAD

**Issue by andrewspiteri** [autofac/Autofac#1229]
The following code used to work prior to v6 upgrade. Now it throws an error as per below:

` private void LifetimeScope_ChildLifetimeScopeBeginning(object? sender, Autofac.Core.Lifetime.LifetimeScopeBeginningEventArgs e)
        {
            if (ReferenceEquals(EnvironmentOptions.CoreLifetimeScopeTag, e.LifetimeScope.Tag))
            {
                try
                {
                    _configuration = e.LifetimeScope.Resolve<Configuration>();
                    _configuration.OnConfigurationChanged += _configuration_OnConfigurationChanged;

                }
                catch (Exception ex)
                {
                    if (_rootLifetimeScope.TryResolve<ILog>(out ILog? log))
                        log?.Error("Error resolving configuration", ex);
                }
                finally
                {
                    _rootLifetimeScope.ChildLifetimeScopeBeginning -= LifetimeScope_ChildLifetimeScopeBeginning;
                }
            }
        }`

`Resolve Operation Starting
{
  Resolve Request Starting
  {
    Service: HomeMat.Core.Logging.Configuration
    Component: HomeMat.Core.Logging.Configuration

    Pipeline:
    -> CircularDependencyDetectorMiddleware
      -> ScopeSelectionMiddleware
        -> SharingMiddleware
          -> RegistrationPipelineInvokeMiddleware
            -> ActivatorErrorHandlingMiddleware
            X- ActivatorErrorHandlingMiddleware
          X- RegistrationPipelineInvokeMiddleware
        X- SharingMiddleware
      X- ScopeSelectionMiddleware
    X- CircularDependencyDetectorMiddleware
  }
  Resolve Request FAILED
    Autofac.Core.DependencyResolutionException: The Activation phase of the resolve pipeline did not populate the request context's Instance.
       at Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
}
Operation FAILED
  Autofac.Core.DependencyResolutionException: An exception was thrown while activating HomeMat.Core.Logging.Configuration.
   ---> Autofac.Core.DependencyResolutionException: The Activation phase of the resolve pipeline did not populate the request context's Instance.
     at Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
     --- End of inner exception stack trace ---
     at Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
     at Autofac.Core.Resolving.Pipeline.ResolvePipelineBuilder.<>c__DisplayClass14_0.<BuildPipeline>b__1(ResolveRequestContext ctxt)
     at Autofac.Core.Pipeline.ResolvePipeline.Invoke(ResolveRequestContext ctxt)
     at Autofac.Core.Resolving.Middleware.RegistrationPipelineInvokeMiddleware.Execute(ResolveRequestContext context, Action`1 next)
     at Autofac.Core.Resolving.Pipeline.ResolvePipelineBuilder.<>c__DisplayClass14_0.<BuildPipeline>b__1(ResolveRequestContext ctxt)
     at Autofac.Core.Resolving.Middleware.SharingMiddleware.<>c__DisplayClass5_0.<Execute>b__0()
     at Autofac.Core.Lifetime.LifetimeScope.CreateSharedInstance(Guid id, Func`1 creator)
     at Autofac.Core.Lifetime.LifetimeScope.CreateSharedInstance(Guid primaryId, Nullable`1 qualifyingId, Func`1 creator)
     at Autofac.Core.Resolving.Middleware.SharingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
     at Autofac.Core.Resolving.Pipeline.ResolvePipelineBuilder.<>c__DisplayClass14_0.<BuildPipeline>b__1(ResolveRequestContext ctxt)
     at Autofac.Core.Resolving.Pipeline.ResolvePipelineBuilder.<>c__DisplayClass14_0.<BuildPipeline>b__1(ResolveRequestContext ctxt)
     at Autofac.Core.Resolving.Middleware.CircularDependencyDetectorMiddleware.Execute(ResolveRequestContext context, Action`1 next)
     at Autofac.Core.Resolving.Pipeline.ResolvePipelineBuilder.<>c__DisplayClass14_0.<BuildPipeline>b__1(ResolveRequestContext ctxt)
     at Autofac.Core.Pipeline.ResolvePipeline.Invoke(ResolveRequestContext ctxt)
     at Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, ResolveRequest request)
     at Autofac.Core.Resolving.ResolveOperation.ExecuteOperation(ResolveRequest request)
`

Note that I have a registration source, that dynamically provides configuration objects, as per below:

`   class ConfigurationRegistrationSource
        : IRegistrationSource
    {
        private readonly ConcurrentDictionary<Type, IComponentRegistration> _registrations;

        public ConfigurationRegistrationSource()
        {
            _registrations = new ConcurrentDictionary<Type, IComponentRegistration>();
        }
        public bool IsAdapterForIndividualComponents => false;

        public IEnumerable<IComponentRegistration> RegistrationsFor(Service service, Func<Service, IEnumerable<ServiceRegistration>> registrationAccessor)
        {
            IServiceWithType? swt = service as IServiceWithType;

            if (swt == null || !typeof(IConfigurationObject).IsAssignableFrom(swt.ServiceType))
            {
                return Enumerable.Empty<IComponentRegistration>();
            }

            return _registrations.GetOrAdd(swt.ServiceType, (t) => CreateServiceRegistration(t, service))
                                 .AsEnumerable();
        }
        
        private IComponentRegistration CreateServiceRegistration(Type forType, Service service)
        {
            IServiceWithType? swt = service as IServiceWithType;

            if (swt is null) throw new ArgumentNullException("Unable to determine service type");

            ComponentRegistration registration = new ComponentRegistration(
                    Guid.NewGuid(),
                    new ConfigurationInstanceActivator(swt.ServiceType),
                    new RootScopeLifetime(),
                    InstanceSharing.Shared,
                    InstanceOwnership.ExternallyOwned,
                    new Service[] { service },
                    new Dictionary<string, object?>());

            return registration;
        }
    }

    sealed class ConfigurationInstanceActivator
        : IInstanceActivator
    {
         
        public ConfigurationInstanceActivator(Type forType)
        {
            LimitType = forType;
        }
        public Type LimitType { get; private set; }

        public object ActivateInstance(IComponentContext context, IEnumerable<Parameter> parameters)
        {
            var configService = context.Resolve<ConfigurationService>();

            var configObject = configService.GetConfiguration(LimitType);

            if (configObject is null) throw new InvalidOperationException($"Unable to find relevant configuration for {LimitType.FullName}");
            
            return configObject;
        }

        public void ConfigurePipeline(IComponentRegistryServices componentRegistryServices, IResolvePipelineBuilder pipelineBuilder)
        {
             
        }

        public void Dispose()
        {
             
        }
    }`

**Comment by alistairjevans** [comment]
Check the "what's changed" guide for 6.0: 
https://autofaccn.readthedocs.io/en/latest/whats-new/upgradingfrom5to6.html

TLDR, activators have changed. You need to implement ConfigurePipeline to add a pipeline stage that does the activation. ActivateInstance isn't called automatically any more.

Check out the `DelegateActivator` in our codename for an example.

**Comment by tillig** [comment]
I hate to ask, but can you clean the code and text up a bit so we can read it? Instead of using single-backticks `which is good for inline code but not multiple lines` instead use triple-backtick fences, like:

````
```c#
public class Foo
{
}
```
````

See that `c#` at the end - everything inside the fences will show as C#. If it's plain text, like an exception message, leave it off:

````
```
Exception message here.
Multiple lines
```
````

**Comment by tillig** [comment]
Or, like, do what @alistairjevans said. ;)

**Comment by andrewspiteri** [comment]
Thank you! 

I ended up using a different approach. 

Will close the issue.
